Mathlib/Combinatorics/SimpleGraph/Walk.lean

```diff
@@ -1173,6 +1176,13 @@ theorem exists_boundary_dart {u v : V} (p : G.Walk u v) (S : Set V) (uS : u ∈
     simp only [getVert_cons_succ, tail_cons]
     exact getVert_copy q n (getVert_zero q).symm rfl
 
+lemma getVert_mem_tail_support {u v : V} (p : G.Walk u v) (hp : ¬p.Nil) (i : ℕ) (hi : 0 < i) :
```

Comment:
```suggestion
lemma getVert_mem_tail_support {u v : V} (p : G.Walk u v) (hp : ¬p.Nil) (i : ℕ) (hi : i ≠ 0) :
```
Non-zero is usually preferred as a hypothesis, and I think it should work with no proof change, but if not you can use `Nat.pos_of_ne_zero hi`